app/tests/components/Dashboard/Dashboard.test.tsx

```diff
@@ -172,12 +173,67 @@ describe('The Dashboard', () => {
     };
     componentTestingHelper.loadQuery(payload);
     componentTestingHelper.renderComponent();
+
     expect(screen.getByText('CCBC-010004')).toBeInTheDocument();
     expect(screen.getByText('submitted')).toBeInTheDocument();
     expect(screen.getByText('Edit')).toBeInTheDocument();
     expect(screen.getByTestId('withdraw-btn-test')).toBeInTheDocument();
   });
 
+  it('Calls the correct mutation when the withdraw button is clicked', async () => {
+    const payload = {
+      Query() {
+        return {
+          allApplications: {
+            nodes: [
+              {
+                id: 'WyJhcHBsaWNhdGlvbnMiLDJd',
+                rowId: 2,
+                owner: '4e0ac88c-bf05-49ac-948f-7fd53c7a9fd6',
+                referenceNumber: 1,
+                status: 'submitted',
+                projectName: null,
+                ccbcId: 'CCBC-010005',
+                lastEditedPage: '',
+                intakeByIntakeId: {
+                  ccbcIntakeNumber: 1,
+                  closeTimestamp: '2024-09-09T13:49:23.513427-07:00',
+                  openTimestamp: '2022-07-25T00:00:00-07:00',
+                },
+              },
+            ],
+          },
+        };
+      },
+    };
+    componentTestingHelper.loadQuery(payload);
+    const user = userEvent.setup();
+
+    componentTestingHelper.renderComponent();
+    expect(screen.getByText('CCBC-010005')).toBeInTheDocument();
+    expect(screen.getByText('submitted')).toBeInTheDocument();
+    expect(screen.getByText('Edit')).toBeInTheDocument();
+    expect(screen.getByTestId('withdraw-btn-test')).toBeInTheDocument();
```

Comment:
using test ids should be a [last resort](https://testing-library.com/docs/queries/bytestid), we should use [`getByRole`](https://testing-library.com/docs/queries/byrole) instead